app/main.py

```diff
@@ -1,34 +1,86 @@
 
 
 class Ship:
-    def __init__(self, start, end, is_drowned=False):
-        # Create decks and save them to a list `self.decks`
-        pass
+    def __init__(self, start: tuple[int, int], end: tuple[int, int]) -> None:
+        self.decks = self._create_decks(start, end)
+        self.is_drowned = False
 
-    def get_deck(self, row, column):
-        # Find the corresponding deck in the list
-        pass
+    def _create_decks(self, start: tuple[int, int],
+                      end: tuple[int, int]) -> list[Deck]:
+        decks = []
+        start_row, start_col = start
+        end_row, end_col = end
 
-    def fire(self, row, column):
-        # Change the `is_alive` status of the deck
-        # And update the `is_drowned` value if it's needed
-        pass
+        if start_row == end_row:
+            for col in range(start_col, end_col + 1):
+                decks.append(Deck(start_row, col))
+        elif start_col == end_col:
+            for row in range(start_row, end_row + 1):
+                decks.append(Deck(row, start_col))
+        return decks
+
+    def get_deck(self, row: int, column: int) -> Deck | None:
+        for deck in self.decks:
+            if deck.row == row and deck.column == column:
+                return deck
+        return None
+
+    def fire(self, row: int, column: int) -> bool:
+        deck = self.get_deck(row, column)
+        if deck and deck.is_alive:
+            deck.is_alive = False
+            if all(not d.is_alive for d in self.decks):
+                self.is_drowned = True
+            return True
 
 
 class Battleship:
-    def __init__(self, ships):
-        # Create a dict `self.field`.
-        # Its keys are tuples - the coordinates of the non-empty cells,
-        # A value for each cell is a reference to the ship
-        # which is located in it
-        pass
-
-    def fire(self, location: tuple):
-        # This function should check whether the location
-        # is a key in the `self.field`
-        # If it is, then it should check if this cell is the last alive
-        # in the ship or not.
-        pass
+    def __init__(self,
+                 ships: list[tuple[tuple[int, int], tuple[int, int]]]
+                 ) -> None:
+        self.size = 10
+        self.field = [["~" for _ in range(self.size)]
+                      for _ in range(self.size)]
+        self.ships = [Ship(*ship) for ship in ships]
+        self._place_ships()
+
+    def _get_ship_cells(self, start: tuple[int, int],
+                        end: tuple[int, int]
+                        ) -> list[tuple[int, int]]:
+        cells = []
+        start_row, start_col = start
+        end_row, end_col = end
+        if start_row == end_row:
+            for col in range(start_col, end_col + 1):
+                cells.append((start_row, col))
+        elif start_col == end_col:
+            for row in range(start_row, end_row + 1):
+                cells.append((row, start_col))
+        return cells
```

Comment:
Code duplication: The logic of the `_get_ship_cells` method in the `Battleship` class is very similar to the `_create_decks` method in the `Ship` class. Consider creating a helper function to eliminate this duplication.